### DISCUSSION TITLE
D_search definition in RWALL

### DISCUSSION THREAD

**Discussion by li2qn** [OpenRadioss/OpenRadioss#1535] [Users]
Hi everyone, 

I am trying to create a rigid plane in Radioss use RWAll/Plane. There is a parameter call D_search, and the definition of it is 
"The secondary nodes to a rigid wall can be defined as a group of nodes and/or as nodes initially at a distance less than the distance (Dsearch) from the rigid wall." 
I had hard time of understanding the definition of D_Search. could you guys please help me to understand what is the this parameter used for? why it is called the secondary nodes to a rigid wall? e,g if my rigid wall is at the x,y plane and centered at (0,0,0), what is the D_search should be? Thanks for any help

Best
Lee

**Comment by MarianBulla** [comment]
Hi Lee,

welcome in the OpenRadioss community.

The RigidWall is an artificial and powerful contact entity, which is created during runtime by the solver. So you don't need to create a new part with its nodes and elements. It will be created by OpenRadioss during runtime at time = zero.
It's stiffness is infinite. So it is a kinematic constraint, with all the advantages and limitations. (for more details, please review the theory manual). It can act frictionless or with friction and can have different shapes:
![image](https://github.com/OpenRadioss/OpenRadioss/assets/105412502/f4a2cfe4-0ef7-4d45-b995-56abbfa83e0c)


Regarding the D_Seach parameter, it defines, which nodes are treated as possible candidates for contact.
You can define a group of nodes (e.g. some parts /GRNODE/PART/...) or a group of nodes which are close to this RWall at time = zero.
These nodes will be searched by the solver and add to the gourp of nodes (we call them "secondary nodes" in addition the the "main sorface") of possible contact candidates:
![image](https://github.com/OpenRadioss/OpenRadioss/assets/105412502/77f07ad3-ba8c-4305-be8f-5a5e325e08f8)

Fore more details, please review the solver manual.

Hope it helps you ?

Thanks and best regards,
Marian

**Reply by li2qn** [reply]
Thanks so much. That's very helpful for my learning

**Comment by MarianBulla** [comment]
Hi li2qn,
You're always welcome. 

Have for with OpenRadioss :-)

Thanks and best regards,
Marian
